rero_mef/agents/api.py

```diff
@@ -0,0 +1,249 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of RERO MEF.
+# Copyright (C) 2018 RERO.
+#
+# RERO MEF is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# RERO MEF is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with RERO MEF; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, RERO does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""API for manipulating records."""
+
+from copy import deepcopy
+
+import click
+from flask import current_app
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus
+from invenio_search import current_search
+
+from ..api import Action, ReroMefIndexer, ReroMefRecord, ReroMefRecordError
+from ..utils import add_md5, add_schema
+
+
+class AgentRecord(ReroMefRecord):
+    """Authority Record class."""
+
+    name = None
+    agent = None
+    viaf_source_code = None
+
+    def __init__(self, *args, **kwargs):
+        """Init class."""
+        super().__init__(*args, **kwargs)
+        self.agent = self.name
+
+    @classmethod
+    def create(cls, data, id_=None, delete_pid=False, dbcommit=False,
+               reindex=False, md5=True, **kwargs):
+        """Create a new agent record."""
+        record = super().create(
+            data=data,
+            id_=id_,
+            delete_pid=delete_pid,
+            dbcommit=dbcommit,
+            reindex=reindex,
+            md5=md5,
+            **kwargs
+        )
+        return record
+
+    @classmethod
+    def update_indexes(cls):
+        """Update indexes."""
+        try:
+            index = 'agents_{agent}'.format(agent=cls.agent)
+            current_search.flush_and_refresh(index=index)
+        except Exception as err:
+            current_app.logger.error(
+                'ERROR flush and refresh: {err}'.format(err=err)
+            )
+
+    def update_test_md5(self, data, dbcommit=False, reindex=False):
+        """Update existing record."""
+        return_record = self
+        if data.get('md5'):
+            data = add_md5(data)
+            data_md5 = data.get('md5', 'data')
+            agent_md5 = self.get('md5', 'agent')
+            if data_md5 == agent_md5:
```

Comment:
as `data_md5` and `agent_md5` are no more used, you can merge 3 lines into the if : 
```
if data.get('md5', 'data') == self.get('md5', 'agent'):
```

Reply:
ok